### DISCUSSION TITLE
Improve grid/table view in Specify 7

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5997] [Feature Requests]
> I find the Sp 6 grid design far easier to use:
> 
> ![image](https://github.com/user-attachments/assets/8c0ba77b-b46b-4382-ada5-3cfdb96e6412)
> 
> I have read [this](https://discourse.specifysoftware.org/t/editing-forms-in-specify-7/1557) (by finding the word ‘grid’) but I’m still uncertain as to whether or not grid column widths (as in the ‘grid view’) can be changed in Sp7.
> 
> ![image](https://github.com/user-attachments/assets/d6e6116c-3005-47e9-a24c-d70ee7a85eb8)

[@willem on Speciforum](https://discourse.specifysoftware.org/t/specify7-crashes-with-certain-edits-to-preparation-type-subview-def/1579/5?u=grant)

There should be some mechanism to switch to a classic, more condensed table view that is entirely read-only. I understand the advantage of our current tables, but I agree with Willem that this could be improved.

**Comment by maxpatiiuk** [comment]
The current implementation in Specify 7 is the way it is because it was simpler - we were able to reuse all existing UI elements in the condensed mode.

If we were to follow Specify 6 implementation, it would involve one of:
- Treating each condensed row as read-only (like you proposed)
- Develop a condensed version of text fields, date pickers, pick lists and etc
  - since we had to add support for these in the workbench to some extend, maybe grid view should just render a mini workbench spreadsheet?

**Comment by grantfitzsimmons** [comment]
>since we had to add support for these in the workbench to some extend, maybe grid view should just render a mini workbench spreadsheet?

I don't mind this suggestion at all if we can do so elegantly.
